frontend/app/services/SubscriptionService.scala

```diff
@@ -185,46 +186,44 @@ class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
       zuoraFeatures <- zuoraSoapService.featuresSupplier.get()
       features = featuresPerTier(zuoraFeatures)(joinData.plan, joinData.featureChoice)
       result <- zuoraSoapService.authenticatedRequest(Subscribe(memberId,
-                                            customerOpt,
-                                            tierPlanRateIds(joinData.plan),
-                                            joinData.name,
-                                            joinData.deliveryAddress,
-                                            paymentDelay,
-                                            casId,
-                                            features))
+                                                      customerOpt,
+                                                      tierPlanRateIds(joinData.plan),
+                                                      joinData.name,
+                                                      joinData.deliveryAddress,
+                                                      paymentDelay,
+                                                      casId,
+                                                      features))
     } yield result
 
   def getPaymentSummary(memberId: MemberId): Future[PaymentSummary] = {
     for {
       subscription <- getSubscriptionStatus(memberId)
-      invoiceItems <- zuoraSoapService.query[InvoiceItem](s"SubscriptionId='${subscription.current}'")
+      invoiceItems <- zuoraSoapService.query[InvoiceItem](s"SubscriptionId='${subscription.currentVersionId}'")
     } yield PaymentSummary(invoiceItems)
   }
 
   def getMembershipSubscriptionSummary(memberId: MemberId): Future[MembershipSummary] = {
-
-    val latestSubF = {
-      for (subscriptions <- getSubscriptions(memberId))
-      yield sortSubscriptions(subscriptions).last
-    }
+    val latestSubF = for {
+      (_, subscription) <- accountWithLatestMembershipSubscription(memberId)
```

Comment:
There's quite a bit of redundancy happening behind the scenes here, and it's because we don't have a get subscription by account anymore. Creating one back and using that instead of the heavy `accountWithLatestMembershipSubscription` will cut back on this redundancy.